core/src/main/java/org/apache/iceberg/AllEntriesTable.java

```diff
@@ -82,18 +83,21 @@ public Schema schema() {
 
     private Scan(
         TableOperations ops, Table table, Long snapshotId, Schema schema, Expression rowFilter,
-        boolean caseSensitive, boolean colStats, Collection<String> selectedColumns,
-        ImmutableMap<String, String> options) {
-      super(ops, table, snapshotId, schema, rowFilter, caseSensitive, colStats, selectedColumns, options);
+        boolean ignoreResiduals, boolean caseSensitive, boolean colStats,
+        Collection<String> selectedColumns, ImmutableMap<String, String> options) {
+      super(
```

Comment:
It is ugly but I don't see another way.

Reply:
The list of parameters seems really large and produces a bunch of changes for adding a new param and it's hard to track. Would it make sense to refactor into an object like `TableScanOptions` or `TableScanContext` and create appropriate builders?

Reply:
I think it is worth exploring using a builder here as a follow-up PR. @edgarRd, would you be interested to try this out?

Reply:
I think it's a good idea to do this, but keep in mind that we will still need to use the refinement pattern for both Scan objects and Options, so that existing instances are immutable. It would help reduce the size of changes when we add options to scans, though.

Reply:
Sure, I can take a stab at it.